### PR TITLE
refactor: issue and redeem requests tables

### PR DESCRIPTION
**Pull request by anton-karlovskiy**

### CODE REVIEW

File: src/common/components/bitcoin-links/block-hash.tsx [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
   blockHash: string;
 }> {
   render(): JSX.Element {
     return (
       <InterlayLink
-        href={
-          (constants.BTC_MAINNET ? constants.BTC_EXPLORER_BLOCK_API : constants.BTC_TEST_EXPLORER_BLOCK_API) +
-          this.props.blockHash
-        }
+        href={BTC_ADDRESS_API + this.props.blockHash}
```
Comment (nud3l) [inline src/common/components/bitcoin-links/block-hash.tsx:11]: `BTC_BLOCK_API`

File: src/common/components/bitcoin-links/transaction.tsx [hunk 2]
```diff
@@ -12,11 +12,7 @@ export default class BitcoinTransaction extends React.Component<{
       <div>
         {this.props.txId ? (
           <InterlayLink
-            href={
-              (constants.BTC_MAINNET ?
-                constants.BTC_EXPLORER_TRANSACTION_API :
-                constants.BTC_TEST_EXPLORER_TRANSACTION_API) + this.props.txId
-            }
+            href={BTC_ADDRESS_API + this.props.txId}
```
Comment (nud3l) [inline src/common/components/bitcoin-links/transaction.tsx:15]: BTC_TRANSACTION_API

File: src/config/blockchain.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+
+const BTC_MAINNET = false;
+
+const BTC_EXPLORER_ADDRESS_API = 'https://blockstream.info/address/';
+const BTC_TEST_EXPLORER_ADDRESS_API = 'https://blockstream.info/testnet/address/';
+
+const BTC_ADDRESS_API = BTC_MAINNET ? BTC_EXPLORER_ADDRESS_API : BTC_TEST_EXPLORER_ADDRESS_API;
+
```
Comment (nud3l) [inline src/config/blockchain.ts:8]: I very much like the refactoring. But you need to add the same thing for the blocks and transactions as well. Note that there is a difference with the URLs:

**Blocks**
```
export const BTC_EXPLORER_BLOCK_API = 'https://blockstream.info/block/';
export const BTC_TEST_EXPLORER_BLOCK_API = 'https://blockstream.info/testnet/block/';
```
**Addresses**
```
export const BTC_EXPLORER_ADDRESS_API = 'https://blockstream.info/address/';
export const BTC_TEST_EXPLORER_ADDRESS_API = 'https://blockstream.info/testnet/address/';
```
**Transactions**
```
export const BTC_EXPLORER_TRANSACTION_API = 'https://blockstream.info/tx/';
export const BTC_TEST_EXPLORER_TRANSACTION_API = 'https://blockstream.info/testnet/tx/';
```

Base URL stays the same, so to simplify you could have a base URL const ("https://blockstream.info/"), miannet/testnet difference ("" for mainnet and "testnet/" for testnet") and then just add the different URLs like "tx", "address", and "block".

So I guess in the end, you'd have a `BTC_BLOCK_API`, `BTC_ADDRESS_API` and `BTC_TRANSACTION_API`.

File: src/pages/Application/issue/modal/status-view.tsx [hunk 4]
```diff
@@ -149,11 +149,7 @@ export default function StatusView(props: StatusViewProps): ReactElement {
             <div className='col'>
               <div className='btc-transaction'>
                 <InterlayLink
-                  href={
-                    (constants.BTC_MAINNET ?
-                      constants.BTC_EXPLORER_TRANSACTION_API :
-                      constants.BTC_TEST_EXPLORER_TRANSACTION_API) + props.request.btcTxId
-                  }
+                  href={BTC_ADDRESS_API + props.request.btcTxId}
```
Comment (nud3l) [inline src/pages/Application/issue/modal/status-view.tsx:152]: BTC_TRANSACTION_API

File: src/pages/Application/issue/modal/status-view.tsx [hunk 5]
```diff
@@ -222,12 +218,7 @@ export default function StatusView(props: StatusViewProps): ReactElement {
                   <div className='row mt-3'>
                     <div className='col text-center'>
                       <InterlayLink
-                        href={
-                          (constants.BTC_MAINNET ?
-                            constants.BTC_EXPLORER_TRANSACTION_API :
-                            constants.BTC_TEST_EXPLORER_TRANSACTION_API) +
-                            props.request.btcTxId
-                        }
+                        href={BTC_ADDRESS_API + props.request.btcTxId}
```
Comment (nud3l) [inline src/pages/Application/issue/modal/status-view.tsx:221]: BTC_TRANSACTION_API

File: src/pages/Application/issue/modal/status-view.tsx [hunk 6]
```diff
@@ -273,12 +264,7 @@ export default function StatusView(props: StatusViewProps): ReactElement {
                 <div className='col text-center'>
                   <InterlayLink
                     // TODO: use the transaction wrapper for this link
-                    href={
-                      (constants.BTC_MAINNET ?
-                        constants.BTC_EXPLORER_TRANSACTION_API :
-                        constants.BTC_TEST_EXPLORER_TRANSACTION_API) +
-                        props.request.btcTxId
-                    }
+                    href={BTC_ADDRESS_API + props.request.btcTxId}
```
Comment (nud3l) [inline src/pages/Application/issue/modal/status-view.tsx:267]: BTC_TRANSACTION_API

File: src/pages/Application/redeem/modal/status-view.tsx [hunk 7]
```diff
@@ -91,11 +91,7 @@ export default function StatusView(props: StatusViewProps): ReactElement {
             <div className='col'>
               <div className='btc-transaction'>
                 <InterlayLink
-                  href={
-                    (constants.BTC_MAINNET ?
-                      constants.BTC_EXPLORER_TRANSACTION_API :
-                      constants.BTC_TEST_EXPLORER_TRANSACTION_API) + props.request.btcTxId
-                  }
+                  href={BTC_ADDRESS_API + props.request.btcTxId}
```
Comment (nud3l) [inline src/pages/Application/redeem/modal/status-view.tsx:94]: BTC_TRANSACTION_API

File: src/pages/dashboard/RedeemRequests/index.tsx [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+function RedeemRequests(): JSX.Element {
+  const {
+    polkaBtcLoaded,
+    prices
+  } = useSelector((state: StoreType) => state.general);
+  const { t } = useTranslation();
+  const statsApi = usePolkabtcStats();
+
+  const [totalSuccessfulRedeems, setTotalSuccessfulRedeems] = useState('-');
```
Comment (nud3l) [inline src/pages/dashboard/RedeemRequests/index.tsx:28]: For some reason, it does not show those defaults, but it shows "unable to fetch data"



https://user-images.githubusercontent.com/14966470/115841159-79f2d700-a457-11eb-856f-b887574bc934.mp4

File: src/pages/dashboard/components/btc-relay.tsx [hunk 9]
```diff
@@ -112,25 +112,21 @@ const BtcRelay = ({ linkButton, displayBlockstreamData }: BtcRelayProps): ReactE
                 {blockstreamTip !== '-' && (
                   <StyledLinkData
                     data={t('dashboard.relay.blockstream_verify_link')}
-                    target={
-                      (constants.BTC_MAINNET ?
-                        constants.BTC_EXPLORER_BLOCK_API :
-                        constants.BTC_TEST_EXPLORER_BLOCK_API) + blockstreamTip
-                    }
+                    target={BTC_ADDRESS_API + blockstreamTip}
```
Comment (nud3l) [inline src/pages/dashboard/components/btc-relay.tsx:115]: BTC_BLOCK_API

File: src/pages/dashboard/oracles/oracles.dashboard.page.tsx [hunk 10]
```diff
@@ -1,32 +1,33 @@
...
-          <div>
-            <PageTitle
-              mainTitle={t('dashboard.oracles.oracles')}
-              subTitle={<TimerIncrement />} />
-            <div className='title-line bg-interlayBlue' />
-            <div className='dashboard-graphs-container'>
-              <OracleStatus />
-            </div>
-            <OracleTable planckLocked='1' />
+    <MainContainer
```
Comment (nud3l) [inline src/pages/dashboard/oracles/oracles.dashboard.page.tsx:15]: Oracle table is also having the switch with rows and columns

image

File: src/pages/dashboard/vaults/vaults.dashboard.page.tsx [hunk 11]
```diff
@@ -1,36 +1,37 @@
...
-              mainTitle={t('dashboard.vault.vaults')}
-              subTitle={<TimerIncrement />} />
-            <div className='title-line bg-interlayBlue' />
-            <div className='vaults-graphs-container dashboard-graphs-container'>
-              <ActiveVaults />
-              <CollateralLocked />
-              <Collateralization />
-            </div>
-            <VaultTable></VaultTable>
+    <MainContainer
```
Comment (nud3l) [inline src/pages/dashboard/vaults/vaults.dashboard.page.tsx:17]: Vaults are also row/column switched:

image

File: src/pages/dashboard/parachain/parachain.dashboard.page.tsx [hunk 12]
```diff
@@ -106,42 +102,43 @@ export default function ParachainDashboard(): ReactElement {
...
+      )}>
+      <div className='w-3/4'>
+        <div>
+          <PageTitle
+            mainTitle={t('dashboard.parachain.parachain')}
+            subTitle={<TimerIncrement />} />
+          <hr className='border-interlayRose' />
+          <div className='parachain-graphs-container dashboard-graphs-container'>
+            <ParachainSecurity />
+            <ActiveStakedRelayers />
```
Comment (nud3l) [inline src/pages/dashboard/parachain/parachain.dashboard.page.tsx:119]: The relayers are also row/column switched:

image

**Review by nud3l — CHANGES_REQUESTED** [review]

File: src/config/blockchain.ts (hunk 3 above)
Comment (anton-karlovskiy) [reply src/config/blockchain.ts:17]: @nud3l 
Thank you for explaining.

> Base URL stays the same, so to simplify you could have a base URL const ("https://blockstream.info/"), miannet/testnet difference ("" for mainnet and "testnet/" for testnet") and then just add the different URLs like "tx", "address", and "block".

For now, I think I will not put much effort into this as we anticipate https://github.com/interlay/polkabtc-ui/pull/212#discussion_r619005713.

> So I guess in the end, you'd have a BTC_BLOCK_API, BTC_ADDRESS_API and BTC_TRANSACTION_API.

Very clear!

File: src/pages/dashboard/RedeemRequests/index.tsx (hunk 8 above)
Comment (anton-karlovskiy) [reply src/pages/dashboard/RedeemRequests/index.tsx:28]: @nud3l 
According to the codebase, it's as expected.
Besides that, https://beta.polkabtc.io/ works that way.

**Review by nud3l — APPROVED** [review]

**Pull request merged**